### PR TITLE
7562 no password validation message

### LINKED ISSUE #7562: Password not changed when validation fails without informing user

### PR DESCRIPTION
**Pull request by javitonino**
Fixes #7562

### CODE REVIEW

**Comment by javitonino** [comment]
Ready for review, please CR @gfiorav (maybe related to strong passwords, I may have forgot something).

**Comment by gfiorav** [comment]
Do you think this is fixed? https://github.com/CartoDB/cartodb/issues/5736

File: app/controllers/carto/api/organization_users_controller.rb [hunk 1]
```diff
@@ -71,12 +71,18 @@ def update
         params_to_update = update_params
 
         # ::User validation requires confirmation
-        params_to_update[:password_confirmation] = params_to_update[:password]
+        password = params_to_update[:password]
+        params_to_update[:password_confirmation] = password
+
+        model_validation_ok = @user.valid?
+        if password.present?
+          model_validation_ok &= @user.valid_password?(:password, password, password)
```
Comment (gfiorav) [inline app/controllers/carto/api/organization_users_controller.rb]: `&&=` ?

File: app/controllers/admin/organization_users_controller.rb [hunk 2]
```diff
@@ -116,6 +120,12 @@ def update
     @user.twitter_datasource_enabled = attributes[:twitter_datasource_enabled] if attributes[:twitter_datasource_enabled].present?
     @user.soft_twitter_datasource_limit = attributes[:soft_twitter_datasource_limit] if attributes[:soft_twitter_datasource_limit].present?
 
+    model_validation_ok = @user.valid?
+    if attributes[:password].present? || attributes[:password_confirmation].present?
+      model_validation_ok &= @user.valid_password?(:password, attributes[:password], attributes[:password_confirmation])
+    end
+    raise Sequel::ValidationFailed.new('Validation failed') unless model_validation_ok
```
Comment (gfiorav) [inline app/controllers/admin/organization_users_controller.rb]: Why use the sequel validation instead of AR?

**Comment by gfiorav** [comment]
Made some comments!

File: app/controllers/admin/organization_users_controller.rb (hunk 2 above)
Comment (javitonino) [inline app/controllers/admin/organization_users_controller.rb]: `@user` is a Sequel model here, and the `Sequel::ValidationFailed` is already rescued below (raised from `@user.save`).

File: app/controllers/carto/api/organization_users_controller.rb (hunk 1 above)
Comment (javitonino) [inline app/controllers/carto/api/organization_users_controller.rb]: Sure :+1:

**Comment by javitonino** [comment]
Just checked that this also fixes #5736

File: app/controllers/admin/organization_users_controller.rb [hunk 3]
```diff
@@ -116,6 +120,12 @@ def update
     @user.twitter_datasource_enabled = attributes[:twitter_datasource_enabled] if attributes[:twitter_datasource_enabled].present?
     @user.soft_twitter_datasource_limit = attributes[:soft_twitter_datasource_limit] if attributes[:soft_twitter_datasource_limit].present?
 
+    model_validation_ok = @user.valid?
+    if attributes[:password].present? || attributes[:password_confirmation].present?
+      model_validation_ok &&= @user.valid_password?(:password, attributes[:password], attributes[:password_confirmation])
```
Comment (gfiorav) [inline app/controllers/admin/organization_users_controller.rb]: Line is too long


😛

**Comment by gfiorav** [comment]
If you tested it and you're confident, 📦 it

**Pull request merged**